dspace-spring-rest/src/test/java/org/dspace/app/rest/csv/CsvImportIT.java

```diff
@@ -284,10 +268,14 @@ private void performImportScript(String[] csv) throws Exception {
         for (String csvLine : csv) {
             out.write(csvLine + "\n");
         }
+
         out.flush();
         out.close();
         out = null;
 
-        runDSpaceScript("metadata-import", "-f", "test.csv", "-e", "admin@email.com", "-s");
+        DSpaceCSV dspaceCsv = new DSpaceCSV(new File(filename), context);
```

Comment:
Was there a specific reason why this integration test cannot use the DSpace CLI command `metadata-import` anymore? 

I know that functionally this is the same thing. But I think that preferably our integration tests should be as [black-box](https://en.wikipedia.org/wiki/Black-box_testing) as possible (e.g. They should not need to know that they have to use the `MetadataImport` class. They should use REST API calls and CLI commands as much as possible).

Reply:
@benbosman : I'd agree with @tomdesair here. Is there a reason to change this one line?  If possible, I'd also prefer this IT calling the DSpace CLI command directly.  If there's some reason it *cannot* do so, that'd be good to understand, so that we can either document it (as a comment) or figure out if there's an underlying bug with ITs calling the DSpace CLI.

Reply:
This can be reverted if preferred. Neither method leaves the browse index untouched.
The CSV import is the problem which causes the `org.dspace.app.rest.BrowsesResourceControllerIT` to fail in Travis.

The reason why the CSV import breaks the BrowsesResourceControllerIT is not part of the Entities codebase scope.

Would you want this line to be reverted?

Reply:
@benbosman : I don't entirely understand your comment here.  I don't understand why using `new DSpaceCSV` instead of `runDSpaceScript` causes this to work better.

I'd prefer the `runDSpaceScript` route (as it's more of an Integration Test).  However, it obviously should *clean up after itself* (including the Browse Index) so that it doesn't affect other ITs.  It looks like `CsvImportIT` is already cleaning up after itself (including Browse index) in it's `destroy()` though: https://github.com/DSpace/DSpace/blob/0367f39a323d3a46def1c00eadef491adf934ee8/dspace-spring-rest/src/test/java/org/dspace/app/rest/csv/CsvImportIT.java#L63

So, if using `runDSpaceScript` is causing different errors, that implies that something else isn't being cleaned up properly -- and it'd be good to try to understand what that is. (It actually might be important enough that it's showing a bug in either our code or in other ITs.)

Reply:
@tdonohue I did test it in more detail, and it didn't completely match what I understood at first either.
When changing https://github.com/atmire/DSpace/commit/155980f1fd77a0840695a26a3e82b5e441857f26 it results in an error https://travis-ci.org/atmire/DSpace/builds/505797454#L5673

This error also occurs locally. I assume the `runDSpaceScript` impacts the context somehow, but didn't verify in the codebase what happens exactly

Reply:
@benbosman : part of the issue may be that you are turning off all Context authorization checks here: https://github.com/DSpace/DSpace/blob/4230762fb516e33ca73b06fc1282f45f0435fded/dspace-spring-rest/src/test/java/org/dspace/app/rest/csv/CsvImportIT.java#L89

It looks like the current code never turns authorization checks back *on*.  So, if I'm reading the code correctly, the line `DSpaceCSV dspaceCsv = new DSpaceCSV(new File(filename), context);` looks to run with a Context that has all authorization checks disabled.

I'm not sure if this is the only problem here, but I think it still points to the fact that the original `runDSpaceScript` is quite different from your replacement code.  The `runDSpaceScript` command would be running on a fresh Context, authorized as a specific user (`admin@email.com`).  Your new code is reusing an existing Context (with authorization accidentally disabled).

It's also possible that you are missing a `context.commit();` somewhere after you create your test data.  Remember, test data won't get written to the test database until a commit occurs.  While commits are automated, sometimes you need to force them to occur in ITs in order for the data to be written _immediately_ to the test database.

Overall, I think this error is pointing at a problem in your tests. I don't think we should bypass it by reusing the Context, but instead find a way to fix the problem in your tests.